examples/function.jl

```diff
@@ -0,0 +1,27 @@
+using KernelDensity, AbstractPlotting, Makie
+import AbstractPlotting: convert_arguments, PlotFunc, plottype
+to_tuple(t::Tuple) = t
+to_tuple(t) = (t,)
+
+
+function convert_arguments(P::PlotFunc, f::Function, args...; kwargs...)
+    tmp = f(args...; Iterators.filter(t -> last(t) != automatic, kwargs)...) |> to_tuple
+    convert_arguments(P, tmp...)
+end
+
+# remove convert_arguments(P, f, x) = (x, f.(x))
+function convert_arguments(P::PlotFunc, d::KernelDensity.UnivariateKDE)
+    ptype = plottype(P, Lines) # choose the more concrete one
+    ptype => convert_arguments(ptype, d.x, d.density)
+end
+
+function convert_arguments(P::Type{<: Combined{T}}, d::KernelDensity.BivariateKDE) where T
+    ptype = plottype(P, Heatmap)
```

Comment:
I'm not sure I understand  `plottype`. I though it would somehow figure out how to combine the plot passed as a function and the automatic plot, but it doesn't always work, for example `plottype(Lines, Lines) != Lines`

Reply:
yeah maybe I shouldn't overload this functionality for `plottype` - it basically just figures out the most concrete plot type... so if you have `plottype(Combined{Any}, Lines) == Lines`, but when it's `plottype(Scatter, Lines) == Scatter)`

Reply:
Then maybe it'd be nice to have a `to_pair(P, args)` where `P` is the plottype that comes explicitly from the user that does the right thing (creates a pair plottype => args), taking into account whether `args` itself is a pair or not. And maybe even a `to_pair(args) = to_pair(Combined{Any, <:Any}, args)`

Reply:
Ok, maybe the name `to_pair` throws me off... I guess you're talking about the possibility of recursively applying convert_arguments? If so, can you come up with an example, were you'd want to use convert_arguments recursively and not directly return the result?

Reply:
It was actually in an attempt to implement a "basic recipe" for the `SeriesVector`, where each entry was basically a collection of attributes and the return type of `convert_arguments`. The logic becomes much clearer if I know that each of those is a pair (pretty much like we do `to_node` to support both `Node` and anything else by converting everything to a `Node`). OTOH the name `to_pair` is not perfect, but I can't think of a much better one...

Here for example I find `ptype => convert_arguments(ptype, d.x, d.density)` a bit puzzling. What if the second call also returns a `Pair`? Do you end up with `ptype => P => args`?

>  If so, can you come up with an example, were you'd want to use convert_arguments recursively and not directly return the result?

With `Grouping` it's quite common as I'd split the data in separate groups, call `convert_arguments` in all the groups and then I'd like to figure out what the overall plottype is and create a structure with only the actual plot arguments. This could maybe be handled by creating a recipe for:

```
plot!(::Combined{T, Tuple{Pair{<:PlotFunc, S}}})
```

so that in the end it doesn't matter if I pass the pair or the arguments.